packages/interaction-kit/src/commands/options/subcommand-group.ts

```diff
@@ -0,0 +1,51 @@
+import type { APIApplicationCommandSubcommandGroupOption } from "discord-api-types/v10";
+import { ApplicationCommandOptionType } from "discord-api-types/v10";
+import type { BaseOptionArgs } from "./option";
+import { Option } from "./option";
+import type Subcommand from "./subcommand";
+
+type SubcommandGroupArgs = {
+	subcommands: Subcommand[];
+} & BaseOptionArgs;
+
+export default class SubcommandGroup extends Option {
+	public readonly type = ApplicationCommandOptionType.SubcommandGroup;
+	public readonly subcommands: Map<string, Subcommand>;
+
+	constructor({ name, description, subcommands }: SubcommandGroupArgs) {
+		super({
+			type: ApplicationCommandOptionType.SubcommandGroup,
+			name,
+			description,
+		});
+
+		this.subcommands = new Map(subcommands.map((cmd) => [cmd.name, cmd]));
+	}
+
+	serialize(): APIApplicationCommandSubcommandGroupOption {
+		const payload =
+			super.serialize() as APIApplicationCommandSubcommandGroupOption;
+
+		payload.options = Array.from(this.subcommands.values()).map((subcommand) =>
```

Comment:
I am ~~groot~~ speed
```suggestion
		payload.options = [...this.subcommands.values()].map((subcommand) =>
```

Reply:
Is this faster now? I thought it was the inverse

Reply:
![Google Chrome Dev - 2022-07-25 at 21 17 13@2x](https://user-images.githubusercontent.com/17960496/180846339-e8713ea4-6c7a-4911-b6a6-569a10e665f8.png)